pymeasure/instruments/instrument.py

```diff
@@ -140,43 +141,53 @@ def control(get_command, set_command, docs,
         :param check_get_errors: Toggles checking errors after getting
         """
 
+        inverse = None
         if map_values and isinstance(values, dict):
             # Prepare the inverse values for performance
             inverse = {v: k for k, v in values.items()}
 
         def fget(self):
+            values_local = values
```

Comment:
for what purpose do you introduce these additional two variables? isn't this redundant?

Reply:
I thought so as well, but it seems python doesn't allow using the global variable `values` if there is a possibility that it is assigned locally (it starts complaining that it is referenced before assignment).

Reply:
I suspected that. Probably this could be avoided if the "dynamic" lookup is handled in an additional else-clause, before the ValueError fallthrough. We would have to check if the structure will be simpler in the end, though, and confirm that the end result/behaviour is the same.

Reply:
I thought further on this, I don't see a way how we end up with a structure that is nice than what is already implemented.